src/vs/editor/browser/controller/mouseHandler.ts

```diff
@@ -294,8 +298,13 @@ export class MouseHandler extends ViewEventHandler implements IDisposable {
 		} else if (targetIsViewZone) {
 			let viewZoneData = <editorBrowser.IViewZoneData>t.detail;
 			if (this.viewHelper.shouldSuppressMouseDownOnViewZone(viewZoneData.viewZoneId)) {
+				focus();
+				this._mouseDownOperation.start(t.type, e);
 				e.preventDefault();
 			}
+		} else if (targetIsWidget && (<string>t.detail).indexOf('codeLensWidget') === 0) {
```

Comment:
This is a layer breaker -- the editor core should not know about specific widgets. We perhaps need something similar to `shouldSuppressMouseDownOnViewZone` but with content widgets